lib/hexo/post.js

```diff
@@ -383,6 +383,25 @@ class Post {
     if (typeof data.disableNunjucks === 'boolean') disableNunjucks = data.disableNunjucks;
 
     const cacheObj = new PostRenderEscape();
+    // data.source does not exist when tag plugins call the markdown renderer
+    const isPost = !data.source || ['html', 'htm'].includes(ctx.render.getOutput(data.source));
```

Comment:
Since we are returning early for non-post files, we only have to initialize a `PostRenderEscape` instance when it is indeed a post.